lib/widgets/fields/padding_field.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_quest/widgets/fields/padding_types.dart';
+
+class PaddingField extends StatefulWidget {
+  final void Function(EdgeInsets) onChanged;
+
+  const PaddingField({
+    super.key,
+    required this.onChanged,
+  });
+
+  @override
+  State<PaddingField> createState() => _PaddingFieldState();
+}
+
+class _PaddingFieldState extends State<PaddingField> {
+  int selectedOption = 1;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            CustomRadioButton(
+              text: "All",
+              index: 1,
+              selectedOption: selectedOption,
+              onTap: (index) {
+                setState(() {
+                  selectedOption = index;
+                });
+              },
+            ),
+            CustomRadioButton(
+              text: "Only",
+              index: 2,
+              selectedOption: selectedOption,
+              onTap: (index) {
+                setState(() {
+                  selectedOption = index;
+                });
+              },
+            ),
+            CustomRadioButton(
+              text: "Symmetric",
+              index: 3,
+              selectedOption: selectedOption,
+              onTap: (index) {
+                setState(() {
+                  selectedOption = index;
+                });
+              },
+            ),
+          ],
+        ),
+        Column(
+          children: [
+            (selectedOption == 1)
+                ? PaddingTypes(
+                    paddingType: PaddingTypeSelection.all,
+                    onChanged: (padding) => {
+                          widget.onChanged(padding),
+                        })
+                : const SizedBox(),
+            (selectedOption == 2)
+                ? PaddingTypes(
+                    paddingType: PaddingTypeSelection.only,
+                    onChanged: (padding) => {
+                          widget.onChanged(padding),
+                        })
+                : const SizedBox(),
+            (selectedOption == 3)
+                ? PaddingTypes(
+                    paddingType: PaddingTypeSelection.symmetric,
+                    onChanged: (padding) => {
+                          widget.onChanged(padding),
+                        })
+                : const SizedBox(),
+          ],
+        ),
+      ],
+    );
+  }
+}
+
+class CustomRadioButton extends StatefulWidget {
```

Comment:
Try to always avoid `Custom` word in naming anything. As this doesn't provide any information about what it is. Instead the name should suggest exactly what custom functionality we are implementing here. Maybe PaddingSelectionButton can be a more appropriate name that provides better context.

also, always keep in consideration which widgets are private and which are public. I am pretty sure that this widget is only going to be used in this file so it should be a private widget.